metabase/metabase
# Add a way to pause/resume and reset each authentication method

Epic https://github.com/metabase/metabase/issues/19428

Adds a menu to enable/disable and reset of each built-in authentication method - Google, LDAP, JWT, SAML.

How to test Google Sign-In:
- Run Metabase
- Go to Admin -> Settings -> Authentication -> Google
- Enter Client ID -> `example.apps.googleusercontent.com` and submit the form
- Go back to the main authentication page, open the menu for the corresponding card
- Pause and resume the auth method. The label in the card should reflect its state.
- Click `Deactivate` in the menu and confirm. It should the corresponding settings.

How to test LDAP:
- Run Metabase
- Run `docker run -p 389:389 osixia/openldap:1.5.0`
- Go to Admin -> Settings -> Authentication -> LDAP
- Enter Host -> `localhost`, Port -> `389`, Username or DN -> `cn=admin,dc=example,dc=org`, Password -> `admin`, User search base -> `dc=example,dc=org` and submit the form
- Go back to the main authentication page, open the menu for the corresponding card
- Pause and resume the auth method. The label in the card should reflect its state.
- Click `Deactivate` in the menu and confirm. It should the corresponding settings.

How to test JWT:
- Run Metabase EE
- Go to Admin -> Settings -> Authentication -> JWT
- Enter URL -> `http://localhost:3000`, click `Generate key`, and submit the form
- Go back to the main authentication page, open the menu for the corresponding card
- Pause and resume the auth method. The label in the card should reflect its state.

How to test SAML:
- Run Metabase EE
- Go to Admin -> Settings -> Authentication -> SAML
- Enter URL -> `https://example.test`, copy and paste certificate contents from https://github.com/metabase/metabase/blob/6dc8bbe7257d81712f189e535779144ff04cd932/test_resources/sso/auth0-public-idp.cert and submit the form
- Go back to the main authentication page, open the menu for the corresponding card
- Pause and resume the auth method. The label in the card should reflect its state.

<img width="543" alt="Screenshot 2022-11-11 at 21 01 38" src="https://user-images.githubusercontent.com/8542534/201412270-0b75b3e4-3c5c-4fa0-aab5-ed0980f773ab.png">
<img width="835" alt="Screenshot 2022-11-11 at 21 01 45" src="https://user-images.githubusercontent.com/8542534/201412281-3e3c19ed-c783-4f34-b6aa-b38738deae5a.png">
